### PR TITLE
Check if the files/dirs from manifest don't exist already when removing them

### LINKED ISSUE #815: Can't run terragrunt apply
I have a simple hello world terragrunt example and I am getting the following error:

```
$ tg apply
[terragrunt] [C:\dev\code\personal\terragrunt-hello-world\live\staging\app] 2019/08/02 15:35:43 Running command: terraform --version
[terragrunt] 2019/08/02 15:35:44 Reading Terragrunt config file at C:/dev/code/personal/terragrunt-hello-world/live/staging/app/terragrunt.hcl
[terragrunt] 2019/08/02 15:35:44 Downloading Terraform configurations from file://C:/dev/code/personal/terragrunt-hello-world/infrastructure into C:/dev/code/personal/terragrunt-hello-world/live/staging/app/.terragrunt-cache/Iuk57wqOPE_7PkKaFmUOii7vrSQ/jIMZ6meZTUEKM4-_y0agCmUjNao
[terragrunt] 2019/08/02 15:35:44 Hit multiple errors:
error downloading 'file://C:/dev/code/personal/terragrunt-hello-world/infrastructure': remove C:\dev\code\personal\terragrunt-hello-world\live\staging\app\.terragrunt-cache\Iuk57wqOPE_7PkKaFmUOii7vrSQ\jIMZ6meZTUEKM4-_y0agCmUjNao\app\myapp\main.tf: The system cannot find the file specified.
[terragrunt] 2019/08/02 15:35:44 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

Steps to reproduct

```
git clone https://github.com/jamie3/terragrunt-hello-world
cd terragrunt-hello-world
cd live/staging/app
# First step works
terragrunt init   
# This fails
terragrunt apply
```

### PR DESCRIPTION
**Pull request by ekini**
Fixes #815

### CODE REVIEW

File: util/file.go [hunk 1]
```diff
@@ -321,7 +321,9 @@ func (manifest *fileManifest) clean(manifestPath string) error {
 				return errors.WithStackTrace(err)
 			}
 		} else {
-			if err := os.Remove(manifestEntry.Path); err != nil {
+			// use RemoveAll here instead of Remove because
+			// the first one doesn't return an error if the file doesn't exist
+			if err := os.RemoveAll(manifestEntry.Path); err != nil {
```
Comment (brikis98) [inline util/file.go]: But it also has the risk of deleting a folder and all its contents. Perhaps the right solution is to ignore "file doesn't exist" errors from `os.Remove`?

Comment (ekini) [reply util/file.go]: I couldn't find another way to ignore this error. I expected `os.ErrNotExist` to work, but it doesn't https://play.golang.org/p/HB4u-EilFCu

There a risk of deleting a folder, but only if https://github.com/gruntwork-io/terragrunt/blob/e8030f6123c3490516202924d00b57c00ecc5692/util/file.go#L84 returns `false` for the directory.

Comment (brikis98) [reply util/file.go]: Yea, we're checking `IsDir` now, but after lots of refactoring, that may get lost. You can use `reflect.TypeOf` and good old `fmt.Sprintf` to print out details of the error so you can detect it correctly.

Comment (ekini) [reply util/file.go]: fmt.Sprintf would make it unportable. Actually, I've checked how RemoveAll works and it uses https://golang.org/pkg/os/#IsNotExist which is exactly what we need!

**Comment by ekini** [comment]
Rebased onto the latest master. The tests seem to be fine still :)

**Comment by brikis98** [comment]
Nice find! I'll kick off tests now.

**Comment by brikis98** [comment]
Tests passed. Merging now!

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.19.17. Thanks @ekini!

**Pull request merged**